ddl/index.go

```diff
@@ -1850,6 +1850,10 @@ func (w *worker) addTableIndex(t table.Table, reorgInfo *reorgInfo) error {
 			if err != nil {
 				break
 			}
+			w.ddlCtx.mu.RLock()
```

Comment:
why here do we need a. read lock?  can we remove lock?

Reply:
This is how we use the hook. There is only one hook instance globally.